sites.md

```diff
@@ -0,0 +1,217 @@
+## Sites API
+With Sites api V4 you can:
+
+- [List Sites](#list-sites)
+- [Create a Site](#create-a-site)
+- [Fetch a Site](#fetch-a-site)
+- [Update a Site](#update-a-site)
+- [Available fields](#available-fields)
+
+
+### List sites
+Fetch a paginated list of sites.
+
+Optional params:
+- `team_id`: if present filter entries by team_id.
+- `updated_after`: if present only return entries updated after given date. Valid values are dates in ISO8601 format.
+- `orderby`: if present allow specify the response order by providing one order clause
+  made of `<field_name> <direction>`. Allowed fields:
+  `updated_at`. Directions: `asc` or `desc`.
+  examples:
+    - `?orderby=updated_at+asc`
+
+```
+GET /api/v4/sites?team_id=26&updated_after=2022-05-24T19:30:30Z&orderby=updated_after+asc&fields=-id,-team_id,-boundaries,-job_number,-address,-site_owner_id,-created_at
+```
+
+```json
+200 OK
+
+[
+  {
+    "uuid": "1b14cac0-bd7f-11ed-9b4e-acde48001122",
+    "name": "Doctor Jack-Jack",
+    "location": {
+      "type": "Point",
+      "coordinates": [
+        174.77604159462516,
+        -41.286319623500816
+      ]
+    },
+    "updated_at": "2023-03-08 20:01:53 +1300"
+  },
+  {
+    "uuid": "f36290bc-bd82-11ed-9b4e-acde48001122",
+    "name": "Captain Ant Claw",
+    "location": {
+      "type": "Point",
+      "coordinates": [
+        77.61010151368907,
+        75.81002131462157
+      ]
+    },
+    "updated_at": "2023-03-07 10:11:12 +1300"
+  }
+]
+```
+
+
+### Create a site
+Creates one site.
+
+Input fields:
+  - **uuid**
+  - **team_id**
+  - **name**
+  - **site_owner_id**
+  - job_number
+  - address
+  - location
+  - boundaries
+
+```json
+POST /api/v4/sites
+
+{
+  "site": {
+    "uuid": "1b14cac0-bd7f-11ed-9b4e-acde48001122",
+    "team_id": 26,
+    "name": "Doctor Jack-Jack",
+    "site_owner_id": 35,
+    "job_number": "5af30ff1ce",
+    "address": "Level 3/354 Lambton Quay, Wellington Central, Wellington 6011",
+    "location": "Point(174.77604159462516 -41.286319623500816)",
+    "boundaries": "POLYGON ((174.7760662374219 -41.28628888757578, 174.77599163848276 -41.2863095461495, 174.77600806701338 -41.286352248914284, 174.77608819800787 -41.2863304566517, 174.7760662374219 -41.28628888757578))"
```

Comment:
@aduuub,

This is the format of the data sent to server to create a site.
On these 2 fields I have used the "recommended" format we have been using for anything that is to be represented in the map. [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry)
Which, as you know, is also what we have been rendering on other serializers to clients to use.

But because I am proposing to expose them as [GeoJSON](https://en.wikipedia.org/wiki/GeoJSON) and because we have been trying to make input and output more similar, I am also wondering if you would prefer to use it for creating sites.

For current mobile implementation I believe you would only need a GeoJson point for `location`. Something like: 
```
{
  "type": "Point",
  "coordinates": [125.6, 10.1]
}
```

Reply:
Adam said there may have no problem having boundaries as a GeoJSON object, but location is better to keep as short as possible to avoid penalising data transfer and storage of large number of Sites.

Reply:
For geometries other than points simple Array or JSON are not enough but it still can take:
Any WKT, which could already cover any existing use-case o the api) or any GeoJSON without losing any of the WKT capabilities.

> Note: GeoJSON additional properties would be discarded when parsing them as inputs and only the geometric information of it would be used.